lib/ingreedy/amount_parser.rb

```diff
@@ -59,7 +59,7 @@ class AmountParser < Parslet::Parser
     private
 
     def word_digits
-      Ingreedy.dictionaries.current.numbers.keys
+      Ingreedy.dictionaries.current.numbers.keys.sort_by(&:length).reverse
```

Comment:
@davidstosik ah yes, I think I wanted to stay away from micro-optimizing as long as the performance specs in global-web were passing. Was following the [pattern here](https://github.com/cookpad/ingreedy/blob/master/lib/ingreedy/root_parser.rb#L135-L140).

Sorting upon initialization sounds like a good idea though, let me take a look